include/revng/ABI/Definition.h

```diff
@@ -379,7 +379,15 @@ class Definition : public generated::Definition {
     }
 
     llvm::SmallVector<model::Register::Values, 8> Result;
-    for (auto Register : GeneralPurposeArgumentRegisters())
+
+    const auto &GPRs = GeneralPurposeArgumentRegisters();
+    const model::Register::Values &RVL = ReturnValueLocationRegister();
+    constexpr model::Register::Values Invalid = model::Register::Invalid;
+    bool NoRVLInGPRs = RVL != Invalid && !llvm::is_contained(GPRs, RVL);
+    if (NoRVLInGPRs && Lookup.contains(RVL))
+      Result.emplace_back(RVL);
```

Comment:
I don't understand what's going on here. What does the return value location has to do with arguments?

Reply:
Return value location is where the "first shadow argument" is placed. Some ABIs (i.e. AArch64) dedicate a register specifically for it (x8 in that case). When going `cabi->raw->cabi` we don't want to lose the fact that the register in question is argument-like.

Reply:
OK, in the snipped of code there's not mention of this.
Mention somewhere `ShadowPointerToAggregateReturnValue`, which is our conventional name for this thing.

Reply:
Added this comment near the snippet:
```cpp
    // Some ABIs (for example, AArch64) designite a special register to be used
    // in the cases where `ShadowPointerToAggregateReturnValue` is necessary.
    // The register behaves in an argument-like fashion, so we want to account
    // for it while sorting argument registers.
```

Feel free to close the thread if it sounds good to you.